File: F-设计模式/6.单件模式/Java2/Singleton.java
package Java;

public class Singleton {
    // 开始就建一个对象，缺点：不是懒创建
    private static Singleton _instance = new Singleton();;

    private Singleton() {};

    public static Singleton getInstance() {
        return _instance;
    }

    public void saySomething() {
        System.out.println("Singleton...");
    }
}

File: E.设计模式/6.单件模式/CPP1/Singleton.cpp
#include "Singleton.h"
#include <iostream>

Singleton* Singleton::_instance = 0;

Singleton* Singleton::instance() {
    if (_instance == 0) {
        _instance = new Singleton();
    }

    return _instance;
}

void Singleton::saySomething() {
    std::cout << "aha..." << std::endl;
}